EventGenerator/src/CeEndpoint_module.cc

```diff
@@ -0,0 +1,140 @@
+// Generates electron at the endpoint energy that will be attached to a mu- in
+// the input SimParticleCollection.   If no suitable muon is found, the
+// filter does not pass the event.
+//
+// Andrei Gaponenko, 2021
+
+#include <iostream>
+#include <string>
+#include <cmath>
+#include <memory>
+
+#include "CLHEP/Vector/ThreeVector.h"
+#include "CLHEP/Vector/LorentzVector.h"
+#include "CLHEP/Random/RandomEngine.h"
+#include "CLHEP/Random/RandExponential.h"
+#include "CLHEP/Units/PhysicalConstants.h"
+
+#include "fhiclcpp/types/Atom.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include "art/Framework/Core/EDFilter.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+
+#include "SeedService/inc/SeedService.hh"
+#include "GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "GlobalConstantsService/inc/ParticleDataTable.hh"
+#include "GlobalConstantsService/inc/PhysicsParams.hh"
+#include "Mu2eUtilities/inc/RandomUnitSphere.hh"
+#include "DataProducts/inc/PDGCode.hh"
+#include "MCDataProducts/inc/StageParticle.hh"
+
+namespace mu2e {
+
+  //================================================================
+  class CeEndpoint : public art::EDFilter {
+  public:
+    struct Config {
+      using Name=fhicl::Name;
+      using Comment=fhicl::Comment;
+      fhicl::Atom<art::InputTag> inputSimParticles{Name("inputSimParticles"),
+          Comment("A SimParticleCollection with input stopped muons.")};
+      fhicl::Atom<std::string> stoppingTargetMaterial{
```

Comment:
I anticipate some code refactoring when more generators are added in the follow up PR.  It will probably include moving these configs to a shared pieced of code.